tests/serverpod_test_server/test/generated_methods/from_json_test.dart

```diff
@@ -0,0 +1,303 @@
+    );
+
+    expect(
+      Types.fromJson({'aBool': null}).aBool,
+      null,
+    );
+  });
+
+  test(
+      'Given non-nullable DateTime, when deserializing with correct value in JSON, then the value is deserialized correctly',
+      () {
+    expect(
+      Types.fromJson({'aDateTime': '2024-01-01T00:00:00.000Z'}).aDateTime,
+      DateTime.tryParse('2024-01-01T00:00:00.000Z'),
+    );
+
+    expect(
+      Types.fromJson(
+        {'aDateTime': DateTime.tryParse('2024-01-01T00:00:00.000Z')},
+      ).aDateTime,
+      DateTime.tryParse('2024-01-01T00:00:00.000Z'),
+    );
+  });
+
+  test(
+      'Given non-nullable Duration, when deserializing with correct value in JSON, then the value is deserialized correctly',
+      () {
+    expect(
+      Types.fromJson({'aDuration': 1000}).aDuration?.inMilliseconds,
+      1000,
+    );
+
+    expect(
+      Types.fromJson({'aDuration': Duration(milliseconds: 1000)})
+          .aDuration
+          ?.inMilliseconds,
+      1000,
+    );
+  });
+
+  test(
+      'Given non-nullable UUID fields, when deserialized from JSON, then the UUID values are deserialized correctly',
+      () {
+    expect(
+      Types.fromJson({'aUuid': '00000000-0000-0000-0000-000000000000'})
+          .aUuid
+          ?.uuid,
+      '00000000-0000-0000-0000-000000000000',
+    );
+
+    expect(
+      Types.fromJson({
+        'aUuid': UuidValue.fromString('00000000-0000-0000-0000-000000000000')
+      }).aUuid?.uuid,
+      '00000000-0000-0000-0000-000000000000',
+    );
+  });
+
+  test(
+      'Given non-nullable Enum, when deserializing with correct value in JSON, then the value is deserialized correctly',
+      () {
+    var types = Types.fromJson({'anEnum': 0});
+    expect(
+      types.anEnum,
+      TestEnum.one,
+    );
+  });
+
+  test(
+      'Given a class with non-nullable Enum, when deserializing with wrong value type in JSON, then a TypeError is thrown',
+      () {
+    expect(
+      () => Types.fromJson({'anEnum': 'one'}),
+      throwsA(isA<TypeError>()),
+    );
+  });
+
+  test(
+      'Given non-nullable Enum, when deserializing with invalid value in JSON, then an ArgumentError is thrown',
+      () {
+    expect(
+      () => Types.fromJson({'anEnum': -1}),
+      throwsA(isA<ArgumentError>()),
+    );
+    expect(
+      () => Types.fromJson({'anEnum': TestEnum.values.length}),
+      throwsA(isA<ArgumentError>()),
+    );
+  });
+
+  test(
+      'Given non-nullable List, when deserializing with wrong value type in JSON, then a TypeError is thrown',
+      () {
+    expect(
+      () => TypesList.fromJson({
+        'anInt': ['test']
+      }),
+      throwsA(isA<TypeError>()),
+    );
+  });
+
+  test(
+      'Given non-nullable List, when deserializing with correct value types in JSON, then values are deserialized correctly',
+      () {
+    var typeList = TypesList.fromJson({
+      'anInt': [1, 2]
+    });
+    expect(typeList.anInt?.length, 2);
+    expect(typeList.anInt?.first, 1);
```

Comment:
Note: Given my previous comment, it is fine to group these expects together as they both relate to the same scenario.

Reply:
Gotcha!